exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/unorderedreceiver/UnorderedReceiverBatch.java

```diff
@@ -149,25 +149,32 @@ private RawFragmentBatch getNextBatch() throws IOException {
     }
   }
 
+  private RawFragmentBatch getNextNotEmptyBatch() throws IOException {
```

Comment:
why is that? this method is used once and doing so will make us duplicate exception handling code. I would appreciate if you provide reason(s) when you make suggestions to avoid the back and forth.

Reply:
No, there is no need to handle `IOException` twice. There are no other methods that throw `IOException`. `SchemaChangeException` is actually never thrown. Please see *TODO* comment and DRILL-2933.